src/interactive.jl

```diff
@@ -0,0 +1,179 @@
+which are not fields of the type, and should be customizable in interactive mode.
+For example, for a constructor `Foo(; x::Bool)`, provide `[x => Bool]`.
+If `T` has fields which should not be customizable, use `NotCustomizable` as the type.
+"""
+customizable(::Type) = ()
+
+function pretty_message(s::AbstractString)
+    replacements = [
+        r"Array{(.*?),1}" => s"Vector{\1}",
+        r"Union{Nothing, (.*?)}" => s"Union{\1, Nothing}",
+    ]
+    return reduce((s, p) -> replace(s, p), replacements; init=s)
+end
+
+"""
+    input_tips(::Type{T}) -> Vector{String}
+
+Provide some extra tips to users on how to structure their input for the type `T`,
+for example if multiple delimited values are expected.
+"""
+input_tips(T::Type{<:Vector}) = ["comma-delimited", input_tips(eltype(T))...]
+input_tips(::Type{Nothing}) = String[]
+input_tips(::Type{Union{T, Nothing}}) where T = ["'nothing' for nothing", input_tips(T)...]
+input_tips(::Type{Secret}) = ["name only"]
+input_tips(::Type) = String[]
+
+"""
+    convert_input(::Type{P}, ::Type{T}, s::AbstractString) -> T
+
+Convert the user input `s` into an instance of `T`.
+A default implementation of `T(s)` exists.
+"""
+convert_input(::Type, T::Type{<:Real}, s::AbstractString) = parse(T, s)
+convert_input(::Type, T::Type, s::AbstractString) = T(s)
+
+function convert_input(P::Type, ::Type{Union{T, Nothing}}, s::AbstractString) where T
+    # This is kind of sketchy because technically, there might be some other input
+    # whose value we want to instantiate with the string "nothing",
+    # but I think that would be a pretty rare occurrence.
+    # If that really happens, they can just override this method.
+    return s == "nothing" ? nothing : convert_input(P, T, s)
+end
+
+function convert_input(::Type, ::Type{Bool}, s::AbstractString)
+    s = lowercase(s)
+    return if startswith(s, "t") || startswith(s, "y")
+        true
+    elseif startswith(s, "f") || startswith(s, "n")
+        false
+    else
+        throw(ArgumentError("Unrecognized boolean response"))
+    end
+end
+
+function convert_input(P::Type, T::Type{<:Vector}, s::AbstractString)
+    startswith(s, "[") && endswith(s, "]") && (s = s[2:end-1])
+    xs = map(strip, split(s, ","))
+    return map(x -> convert_input(P, eltype(T), x), xs)
+end
+
+"""
+    prompt(::Type{P}, ::Type{T}, ::Val{name::Symbol}) -> Any
+
+Prompts for an input of type `T` for field `name` of plugin type `P`.
+Implement this method to customize particular fields of particular types.
+"""
+prompt(P::Type, T::Type, name::Symbol) = prompt(P, T, Val(name))
+
+# The trailing `nothing` is a hack for `fallback_prompt` to use, ignore it.
+function prompt(P::Type, ::Type{T}, ::Val{name}, ::Nothing=nothing) where {T, name}
+    tips = join([T; input_tips(T); "default=$(repr(defaultkw(P, name)))"], ", ")
+    default = defaultkw(P, name)
+    print(pretty_message("Enter value for '$name' ($tips): "))
+    input = strip(readline())
```

Comment:
Consider using `Base.prompt` instead as it can tell the difference between an End-of-Transmission character and just pressing enter:
```julia
julia> input = Base.prompt("demo");
demo:

julia> show(input)
""

julia> input = Base.prompt("demo");
demo: ^D
julia> show(input)
nothing
```

The main advantage here is you could do an early termination of the interactive prompt if you desire. The code would change to something like:

```suggestion
    response = prompt(pretty_message("Enter value for '$name' ($tips)"), default=default)
    response !== nothing && (response = strip(response))
```

Reply:
This is a good idea, I didn't know that function existed. I guess the question is what would we do if we got `nothing`? Do we treat it like a keyboard interrupt and return the user to the REPL?

Reply:
I would expect an EOT to exit the current interactive prompt from PkgTemplates.